File: react_gsea/dataFromResults.py
def dataFromResult(input_set, ranked_entities):
  data = []
  for i, ranked_entity in enumerate(ranked_entities):
    if ranked_entity in input_set:
      data.append({
        'x': i,
        'y': -(i-(len(ranked_entities)/2)) / (len(ranked_entities)/2),
        'b':  1
      })
    else:
      data.append({
        'x': i,
        'y': -(i-(len(ranked_entities)/2)) / (len(ranked_entities)/2),
        'b':  0
      })
  return sorted(data, key=lambda r: r['x'])

File: src/dataFromResults.js
/**
 * Helper for creating ReactGSEA data from enrichment input/output
 */
export function dataFromResults({ input_set, ranked_entities }) {
  var data = new Array();

  for (var i = 0; i < ranked_entities.length; i++) {
    if (input_set.includes(ranked_entities[i])) {

      data.push({
        "x": i,
        "y": -(i - (ranked_entities.length / 2)) / (ranked_entities.length / 2),
        "b": 1
      });
    } else {
      data.push({
        "x": i,
        "y": -(i - (ranked_entities.length / 2)) / (ranked_entities.length / 2),
        "b": 0
      });
    }
  }

  data.sort(function (a, b) {
    return b.x - a.x;
  });

  return data
}
